lib/flags.ml

```diff
@@ -189,6 +189,11 @@ let print_hyps_limit = ref (None : int option)
 let set_print_hyps_limit n = print_hyps_limit := n
 let print_hyps_limit () = !print_hyps_limit
 
+(* Flag for compact display of goals *)
+
+let compact_context = ref false
+let is_compact_context() = !compact_context
```

Comment:
There are enough flags already. Why not to put this (and the `GOption` declaration in `printer.ml`.